src/main/scala/nl/knaw/dans/easy/pid/PidGeneratorServlet.scala

```diff
@@ -19,32 +19,44 @@ import nl.knaw.dans.lib.error._
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 import org.scalatra._
 
-import scala.util.Try
+import scala.util.{ Failure, Success, Try }
 
-class PidGeneratorServlet(app: PidGeneratorApp) extends ScalatraServlet with DebugEnhancedLogging {
+class PidGeneratorServlet(app: PidGeneratorApp, configuration: Configuration) extends ScalatraServlet with DebugEnhancedLogging {
   logger.info("PID Generator Servlet running...")
 
   get("/") {
     contentType = "text/plain"
-    Ok("Persistent Identifier Generator running")
+    Ok(s"Persistent Identifier Generator running (v${ configuration.version })")
   }
 
-  private def respond(result: Try[String]): ActionResult = {
-    result.map(Ok(_))
-      .doIfFailure { case e => logger.error(e.getMessage, e) }
-      .getOrRecover {
-        case e: RanOutOfSeeds => NotFound(e.getMessage)
```

Comment:
this is no longer a possibility?

Reply:
No, unfortunately not... We initialize the seed, but then modify that value as we go. However, if we run out of seeds (e.g. we get back to the initial seed), the application will now detect it as a duplicate PID, since we store all minted PIDs.

Reply:
Sidenote: since we kept the algorithm for calculating the next PID the same, the last possible PID to be generated is the one corresponding to the initial seed!

```
given initial seed s(0)
and seed s(n) as the current seed in the database
and function f that calculates the next seed - f(s(n)) = s(n+1)
and function g that generates a PID from a seed - g(s(n)) = p(n)

we do:
algo(n) = {
  val s' = f(s(n))
  val p = g(s')
  store(s')
  store(p)
  return p
}

therefore the first PID to be generated is p(n+1)

likewise:
if s(z) is the last seed that f can generate before returning to s(0) - e.g. f(s(z)) = s(0),
then we get algo(z) = p(0)
```